superset-frontend/src/views/CRUD/data/dataset/AddDataset/index.tsx

```diff
@@ -71,7 +71,11 @@ export default function AddDataset() {
   >(datasetReducer, null);
 
   const HeaderComponent = () => (
-    <Header setDataset={setDataset} datasetName={dataset?.dataset_name ?? ''} />
+    <Header
+      setDataset={setDataset}
+      title={dataset?.table_name ?? 'New dataset'}
```

Comment:
May want to wrap in a t('New dataset')
Also, may be worth using exported const from previous comment if you want to keep this in sync with what the default title is for header.tsx.

If making the title prop optional like mentioned in comment on Header/index.tsx then this could be simplified to `title={dataset?.table_name}` and the default value 'New dataset' would get applied by Header and schema prop could be removed

Reply:
Updated in [`this commit`](https://github.com/apache/superset/pull/21557/commits/e9f6eb28b1a684b08eb097d43dee99034d775df8)